kmip/services/server/engine.py

```diff
@@ -1509,22 +1509,37 @@ def _process_destroy(self, payload):
         else:
             unique_identifier = self._id_placeholder
 
-        self._get_object_with_access_controls(
+        managed_object = self._get_object_with_access_controls(
             unique_identifier,
             enums.Operation.DESTROY
         )
 
-        # TODO (peterhamilton) Process attributes to see if destroy possible
-        # 1. Check object state. If invalid, error out.
-        # 2. Check object deactivation date. If invalid, error out.
-
-        self._logger.info(
-            "Destroying an object with ID: {0}".format(unique_identifier)
-        )
+        # TODO If in "ACTIVE" state, we need to check its "Deactivation date"
+        # to see whether it can be destroyed or not
+        if hasattr(managed_object, 'state'):
+            if managed_object.state != enums.State.PRE_ACTIVE and \
+               managed_object.state != enums.State.DEACTIVATED and \
+               managed_object.state != enums.State.COMPROMISED:
```

Comment:
It might also just be easier to do an `ACTIVE` check here instead, since that's the only state that can't be directly destroyed.